src/ecpool_worker.erl

```diff
@@ -67,10 +67,18 @@
 %%--------------------------------------------------------------------
 
 %% @doc Start a pool worker.
--spec(start_link(pool_name(), pos_integer(), module(), list()) ->
+-spec(start_link(pool_name(), pos_integer(), module(), list(), {pid(), reference()}) ->
       {ok, pid()} | ignore | {error, any()}).
-start_link(Pool, Id, Mod, Opts) ->
-    gen_server:start_link(?MODULE, [Pool, Id, Mod, Opts], []).
+start_link(Pool, Id, Mod, Opts, {InitialConnectResultReceiverPid, Ref}) ->
+    {ok, Pid} = OkResp = gen_server:start_link(?MODULE, [Pool, Id, Mod, Opts], []),
+    case gen_server:call(Pid, init_connect) of
```

Comment:
no need to turn this into a call, just pass the collector and reference down to the worker's `init/1`, and reply can be sent from the worker.

Reply:
Yes, if we don't care about stoppning this worker (as it will be stopped anyway by the ecpool module after it receive the error message) we can just as well do the initial connect in in the init function and return success.

Reply:
Changed to the initial connect in the init function: [8f8a6d0](https://github.com/emqx/ecpool/pull/50/commits/8f8a6d0e858bbff470128dbd376f1171a7fd00d0)